File: apps/api/plane/license/api/urls/help_center.py
# Copyright (c) 2023-present Plane Software, Inc. and contributors
# SPDX-License-Identifier: AGPL-3.0-only
# See the LICENSE file for details.

from django.urls import path

from plane.license.api.views.help_center import (
    InstanceHelpArticleAssetEndpoint,
    InstanceHelpArticleDetailEndpoint,
    InstanceHelpArticleEndpoint,
    InstanceHelpArticleTranslationEndpoint,
    InstanceHelpCategoryDetailEndpoint,
    InstanceHelpCategoryEndpoint,
    InstanceHelpCenterExportEndpoint,
    InstanceHelpCenterImportEndpoint,
)

# God Mode authoring of the shared (instance-global) Help Center.
# Mounted under /api/instances/ -> /api/instances/help/...
urlpatterns = [
    path(
        "help/categories/",
        InstanceHelpCategoryEndpoint.as_view(http_method_names=["get", "post"]),
        name="instance-help-categories",
    ),
    path(
        "help/categories/<uuid:pk>/",
        InstanceHelpCategoryDetailEndpoint.as_view(http_method_names=["get", "patch", "delete"]),
        name="instance-help-category-detail",
    ),
    path(
        "help/articles/",
        InstanceHelpArticleEndpoint.as_view(http_method_names=["get", "post"]),
        name="instance-help-articles",
    ),
    path(
        "help/articles/<uuid:pk>/",
        InstanceHelpArticleDetailEndpoint.as_view(http_method_names=["get", "patch", "delete"]),
        name="instance-help-article-detail",
    ),
    path(
        "help/articles/<uuid:pk>/translations/<str:locale>/",
        InstanceHelpArticleTranslationEndpoint.as_view(http_method_names=["put", "patch"]),
        name="instance-help-article-translation",
    ),
    path(
        "help/articles/<uuid:pk>/assets/",
        InstanceHelpArticleAssetEndpoint.as_view(http_method_names=["post"]),
        name="instance-help-article-assets",
    ),
    path(
        "help/articles/<uuid:pk>/assets/<uuid:asset_id>/",
        InstanceHelpArticleAssetEndpoint.as_view(http_method_names=["patch"]),
        name="instance-help-article-asset-detail",
    ),
    path(
        "help/export/",
        InstanceHelpCenterExportEndpoint.as_view(http_method_names=["get"]),
        name="instance-help-export",
    ),
    path(
        "help/import/",
        InstanceHelpCenterImportEndpoint.as_view(http_method_names=["post"]),
        name="instance-help-import",
    ),
]

File: packages/types/src/help-center.ts
/**
 * Copyright (c) 2023-present Plane Software, Inc. and contributors
 * SPDX-License-Identifier: AGPL-3.0-only
 * See the LICENSE file for details.
 */

// Shapes for God Mode (instance-admin) authoring of the shared, instance-global
// Help Center. Content is multilingual (VI/EN/KO); the admin chrome is English-only.
// Mirrors the API serializers in plane/app/serializers/help_center.py
// (HelpCategoryAdminSerializer / HelpArticleAdminSerializer).

export type THelpLocale = "vi" | "en" | "ko";
export type THelpArticleStatus = "draft" | "published";

export interface IHelpCategoryTranslation {
  locale: THelpLocale;
  name: string;
}

export interface IHelpArticleTranslation {
  locale: THelpLocale;
  title: string;
  description_html: string;
  description_json: object;
}

export interface IHelpCategory {
  id: string;
  slug: string;
  sort_order: number;
  icon: string;
  color: string;
  is_active: boolean;
  translations: IHelpCategoryTranslation[];
}

export interface IHelpArticle {
  id: string;
  slug: string;
  category: string | null;
  sort_order: number;
  status: THelpArticleStatus;
  translations: IHelpArticleTranslation[];
  created_at: string;
  updated_at: string;
}

// ── Write payloads ───────────────────────────────────────────────────────────

export interface IHelpCategoryTranslationInput {
  locale: THelpLocale;
  name: string;
}

export interface IHelpCategoryCreate {
  translations: IHelpCategoryTranslationInput[];
  icon?: string;
  color?: string;
  is_active?: boolean;
}

export interface IHelpCategoryUpdate {
  icon?: string;
  color?: string;
  is_active?: boolean;
  sort_order?: number;
  translations?: IHelpCategoryTranslationInput[];
}

export interface IHelpArticleTranslationInput {
  locale: THelpLocale;
  title: string;
  description_html?: string;
  description_json?: object;
}

export interface IHelpArticleCreate {
  category: string | null;
  translations: IHelpArticleTranslationInput[];
  status?: THelpArticleStatus;
}

export interface IHelpArticleUpdate {
  category?: string | null;
  slug?: string;
  sort_order?: number;
  status?: THelpArticleStatus;
  translations?: IHelpArticleTranslationInput[];
}

export interface IHelpArticleTranslationUpsert {
  title: string;
  description_html?: string;
  description_json?: object;
}

// ── Bundle export/import ──────────────────────────────────────────────────────

// Result of importing a help-center bundle (.zip) into the current environment.
// Mirrors the API response of POST /api/instances/help/import/.
export type THelpBundleImportResult = {
  categories: number;
  articles: number;
  images: number;
};
